helix-core/src/syntax/span.rs

```diff
@@ -67,6 +71,79 @@ pub fn span_iter(spans: Vec<Span>) -> impl Iterator<Item = HighlightEvent> {
     }
 }
 
+impl SpanIter {
+    fn start_span(&mut self, span: Span) {
+        use HighlightEvent::*;
+
+        debug_assert!(span.start <= span.end);
+        self.event_queue
+            .push_back(HighlightStart(Highlight(span.scope)));
+        self.range_ends.push(span.end);
+    }
+
+    fn process_range_end(&mut self, end: usize) -> HighlightEvent {
+        use HighlightEvent::*;
+
+        let start = replace(&mut self.cursor, end);
+        if start != end {
+            debug_assert!(start < end);
+            self.event_queue.push_back(HighlightEnd);
+            Source { start, end }
+        } else {
+            HighlightEnd
+        }
+    }
+
+    // Any subslices that end before intersect span needs to be subsliced, consume the
+    // left part of the subslice and leave the right.
+    fn partition_spans_at(&mut self, intersect: usize) {
+        let first_partioned_span = self.spans[self.index];
+
+        let mut i = self.index;
+        while let Some(span) = self.spans.get_mut(i) {
+            if span.start != self.cursor || span.end < intersect {
+                break;
+            }
+
+            let mut partinoed_span = *span;
```

Comment:
typos for these two vars on "partitioned"

Reply:
fixed in 20302313b5d274d231c8c2e505a5306601c0173d